lib/OpenQA/Worker/Cache.pm

```diff
@@ -56,6 +55,7 @@ sub set {
 sub init {
     my $class;
     ($host, $location) = @_;
+    $db_file = catdir($location, $db_file);
```

Comment:
this breaks on re-init after api-failures and creates https://progress.opensuse.org/issues/18110